tasks/setup_debian.yml

```diff
@@ -10,26 +10,63 @@
         state: present
         update_cache: true
 
-    - name: Install mariadb repository key
+    - name: Ensure keyrings dir exists
+      ansible.builtin.file:
+        path: /etc/apt/keyrings
+        state: directory
+        owner: root
+        group: root
+        mode: "0755"
+
+    - name: Fetch mariadb repository key
       ansible.builtin.get_url:
-        url: "https://mariadb.org/mariadb_release_signing_key.asc"
-        dest: "/etc/apt/trusted.gpg.d/mariadb.asc"
+        url: https://mariadb.org/mariadb_release_signing_key.asc
+        dest: /etc/apt/keyrings/mariadb-archive-keyring.asc
         # //TEMP todo
         # checksum: sha256:http://example.com/path/sha256sum.txt
         owner: root
         group: root
         mode: "0644"
+      register: fetch_repository_key
+
+    - name: Dearmor mariadb repository key
+      when: fetch_repository_key is changed
+      ansible.builtin.command: >
+        gpg /etc/apt/keyrings/mariadb-archive-keyring.asc
+        --dearmor
+        --output /etc/apt/keyrings/mariadb-archive-keyring.gpg
 
     - name: Determine distribution name (Debian sid pb)
       ansible.builtin.set_fact:
         distribution: "{% if 'n/a' in ansible_distribution_release %}sid{% else %}{{ ansible_distribution_release }}{% endif %}"
 
     - name: Setup mariadb repository sourcelist entry
+      when: not mariadb_use_official_repo_deb822
       ansible.builtin.copy:
         dest: /etc/apt/sources.list.d/mariadb.list
         content: |
           # Ansible managed
-          deb {{ mariadb_use_official_repo_url }}/{{ mariadb_use_official_repo_version }}/{{ ansible_distribution | lower() }} {{ distribution }} main
+          deb \
+          [signed-by=/etc/apt/keyrings/mariadb-archive-keyring.gpg] \
+          {{ mariadb_use_official_repo_url }}/{{ mariadb_use_official_repo_version }}/{{ ansible_distribution | lower() }} \
+          {{ distribution }} \
+          main
+        owner: root
+        group: root
+        mode: "0644"
+
+    - name: Setup mariadb repository sources entry
+      when: mariadb_use_official_repo_deb822
+      ansible.builtin.copy:
+        dest: /etc/apt/sources.list.d/mariadb.sources
```

Comment:
Should the role be responsible for cleaning up the files when changing `mariadb_use_official_repo_deb822` ?

Reply:
Nope, the role should consider a from scratch deployment (cleaning scenarios are too complex IMO).